src/schematools/types.py

```diff
@@ -377,18 +478,25 @@ def fields(self) -> Iterator[DatasetFieldSchema]:
         if self.has_compound_key:
             yield DatasetFieldSchema(_id="id", _parent_table=self, _required=True, type="string")
 
-    def get_fields_by_id(self, field_ids: List[str]) -> Iterator[DatasetFieldSchema]:
-        for field in self.fields:
-            if field.id in set(field_ids):
-                yield field
+    @lru_cache()
+    def get_fields_by_id(self, field_ids: Tuple[str]) -> Iterator[DatasetFieldSchema]:
+        """Get the fields based on the ids of the fields.
 
+        args:
+            field_ids: The ids of the fields.
+            NB. This needs to be a tuple, lru_cache only works on immutable arguments.
+        """
+        return [field for field in self.fields if field.id in set(field_ids)]
```

Comment:
vraagje: gaat Python hier bij iedere `if` ook overnieuw de set overnieuw aanmaken? Ik denk het wel...

Reply:
Yes, it will:

```python
from typing import Iterable, TypeVar

_T = TypeVar("_T")


class MySet(set):
    def __init__(self, iterable: Iterable[_T] = ...) -> None:
        print("So here I am sitting at the bus stop")
        super().__init__(iterable)


numbers = list(range(10))
selected = [5, 7]
print([n for n in numbers if n in MySet(selected)])
```
Output:
```
So here I am sitting at the bus stop
So here I am sitting at the bus stop
So here I am sitting at the bus stop
So here I am sitting at the bus stop
So here I am sitting at the bus stop
So here I am sitting at the bus stop
So here I am sitting at the bus stop
So here I am sitting at the bus stop
So here I am sitting at the bus stop
So here I am sitting at the bus stop
[5, 7]
```